src/Chapter3.hs

```diff
@@ -970,6 +1058,21 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data WeekDay = Sunday|Monday|Tuesday|Wednesday|Thursday|Friday|Saturday deriving (Show,Read,Eq,Ord,Enum)
+
+isWeekend::WeekDay->Bool
+isWeekend d 
+  | d == Sunday || d == Saturday = True
+  | otherwise = False
+
+nextDay :: WeekDay->WeekDay
+nextDay Saturday = Sunday
+nextDay d = toEnum $ fromEnum d + 1
```

Comment:
Nice usage of standard `Enum` functions! Though, this particular function is already implemented and it's called `succ`. Unfortunately, it fails in runtime on the last constructor, that's why you still need to handle it separately.
```suggestion
nextDay d = succ d
```